### DISCUSSION TITLE
import failed on multiple nodes

### DISCUSSION THREAD

**Discussion by georgebeale** [Comfy-Org/ComfyUI#5949] [Q&A]
![import failed](https://github.com/user-attachments/assets/38bf21f7-e20e-4143-b662-41932124e1b8)

I am having an issue where multiple nodes that I am installing via the custom nodes manager are failing to import.  Some import just fine, but others fail, and despite trying multiple solutions I cannot resolve the error.

**Comment by blepping** [comment]
There's no way for anyone to help you without seeing the actual errors that caused the nodes to fail to load. There should be console output with the full exception.

**Reply by georgebeale** [reply]
Here is a quick video of the logs.  This is only regarding the reactor custom node, but I am having this error import failed message for multiple custom nodes.  

https://youtu.be/_4Ljw_lbEpc

**Reply by blepping** [reply]
That's a strange way to share log output! After watching the video, it looks like you issue is building the `insightface` Python package. It's failing because it apparently cannot find Python development while compiling C code that's part of that package (namely header files, but presumably it would also fail trying to link):

![image](https://github.com/user-attachments/assets/d1669db5-3115-4eef-8d5e-39c04fe7e47d)

I don't know why that would cause nodes like undo/redo or upscale to fail. Maybe it's a separate problem, but I didn't see anything specific to the nodes you showed as having an import issue.

Possibly I missed it among all the other output. Since it's a video, no way to do a text search for keywords.

**Comment by ltdrdata** [comment]
![image](https://github.com/user-attachments/assets/f088de26-7d94-4ca7-b4fe-dc04e03339fb)

Do not install `insightface` like a custom_node. First, delete the path `ComfyUI/custom_nodes/insightface`. Then, follow the installation guide for comfyui-reactor-node to properly install insightface.

https://github.com/Gourieff/#installation

**Comment by veeroo18** [comment]
I am getting this error in my manager as well and now it shows following errors after restart ComfyUI
![image](https://github.com/user-attachments/assets/5e011b41-71a4-44e9-ae9f-fa4442caa0ed)

**Comment by ltdrdata** [comment]
> I am getting this error in my manager as well and now it shows following errors after restart ComfyUI
> ![image](https://github.com/user-attachments/assets/5e011b41-71a4-44e9-ae9f-fa4442caa0ed)
> 

You need to resolve your network environment issues.

This is the first time I've encountered a case where PyPI is inaccessible—could it be a closed network?
Another possibility is blocking by a firewall.


If not, you may need to use a VPN or another solution to fix the issue.

**Reply by veeroo18** [reply]
oh ok , 
It was working few hours back, when i tried to install reactor custom node via missing custom node install it went wrong and now its failing. this way.

**Reply by veeroo18** [reply]
I tried using vpn, tested stability on packetstatus and its good - great range. its an open network and disabled firewall . still its giving me same errors. what could be the reason. ?

**Reply by ltdrdata** [reply]
https://github.com/pypa/pip/issues/9508

You need to read that thread.

**Comment by veeroo18** [comment]
strangely comfyui manager shows 500 Internal Server Error Server got itself in trouble
![image](https://github.com/user-attachments/assets/823dff92-9bdd-47f1-980a-c1cfea8d0632)

**Comment by veeroo18** [comment]
ok here is what I did

I went to the terminal of comfy ui 
installed insightface, it showed some error at end mentioning                 
v14+ needed download from here  https://visualstudio.microsoft.com/visual-cpp-build-tools/
downloaded and installed only build tools of 130 mb 
then installed gguf via pip in terminal of comfyui 
and restarted again, 
then enabled nodes in node manager and now its loading installed
